spec/factories/factories.rb

```diff
@@ -157,6 +157,42 @@
     can_choose_topic_to_review true
   end
 
+  factory :assignment_mapping, class: Assignment do
```

Comment:
I think you do not need this factory since there is one factory for assignment already.

Reply:
We tried using that, but the other factory has a hard fix on the sequence number for few of the other existing tests. 

The issue is due to this:
sequence(:name, 2) {|n| n = 2; "final#{n}" }

So, for few of the tests, csv file present on the file storage to check with the expected result has this (final2) hardcoded and the check is against it. 

For our scenario, we needed more than 2 assignments in our database but due to the existing code, we couldn't generate more than 2 assignments.

We added the new factory as the older one was not fitting our case. Let us know if we have to fix the existing ones as I feel the above code was made to just make certain tests pass but is not fit for new tests which would need more than 2 assignments.

Let us know your thought.